plugins/uefi-capsule/fu-uefi-cod-device.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-uefi-cod-device.h"
+#include "fu-uefi-common.h"
+
+struct _FuUefiCodDevice
+{
+	FuUefiDevice		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuUefiCodDevice, fu_uefi_cod_device, FU_TYPE_UEFI_DEVICE)
+
+static gboolean
+fu_uefi_device_get_results (FuDevice *device, GError **error)
+{
+	FuUefiDevice *device_uefi = FU_UEFI_DEVICE (device);
+	const gchar *tmp;
+	fwupd_guid_t guid = { 0x0 };
+	gsize bufsz = 0;
+	guint32 status = 0;
+	guint32 total_size = 0;
+	g_autofree gchar *err_msg = NULL;
+	g_autofree gchar *guidstr = NULL;
+	g_autofree gchar *version_str = NULL;
+	g_autofree guint8 *buf = NULL;
+
+	/* lets assume we just did one update */
+	if (!fu_efivar_get_data (FU_EFIVAR_GUID_EFI_GLOBAL, "Capsule0000",
+				 &buf, &bufsz, NULL, error)) {
+		g_prefix_error (error, "failed to read EFI variable: ");
+		return FALSE;
+	}
+
+	/* sanity check */
+	if (!fu_common_read_uint32_safe (buf, bufsz, 0x00, &total_size, G_LITTLE_ENDIAN, error))
+		return FALSE;
+	if (total_size < 0x3A) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INTERNAL,
+				     "EFI_CAPSULE_RESULT_VARIABLE_HEADER too small");
+		return FALSE;
+	}
+
+	/* verify guid */
+	if (!fu_memcpy_safe (guid, sizeof(guid), 0x0,		/* dst */
+			     buf, bufsz, 0x08, sizeof(guid), error))
+		return FALSE;
+	guidstr = fwupd_guid_to_string (&guid, FWUPD_GUID_FLAG_MIXED_ENDIAN);
+	if (g_strcmp0 (guidstr, fu_uefi_device_get_guid (device_uefi)) != 0) {
```

Comment:
this seems to be asking for a struct definition of what this header format is supposed to be looking like.  then you can do the check with `sizeof(CapsuleResultVariableHeader)` and access from `header->guid` and such.

Reply:
That's not endian clean tho; although I admit in this context LE is probably the only thing we care about.